gender_novels/novel.py

```diff
@@ -137,17 +139,72 @@ def _load_novel_text(self):
             err += "at the expected location ({file_path})."
             raise FileNotFoundError(err)
 
-        # Extract Project Gutenberg Boilerplate
-        if text.find('*** START OF THIS PROJECT GUTENBERG EBOOK') > -1:
-            end_intro_boilerplate = text.find(
-                '*** START OF THIS PROJECT GUTENBERG EBOOK')
-            # second set of *** indicates start
-            start_novel = text.find('***', end_intro_boilerplate + 5) + 3
-            end_novel = text.find('*** END OF THIS PROJECT GUTENBERG EBOOK')
-            text = text[start_novel:end_novel]
+        # These two functions will ensure that the text becomes only the novel's own text,
+        # removing the Project Gutenberg .
+        text = self.remove_boilerplate_text(text)
+        text = self.remove_table_of_contents(text)
+
+        return text
+
+
+    def remove_boilerplate_text(text):
+        """
+        Removes the boilerplate text from an input string of a novel.
+        Currently only supports boilerplate removal for Project Gutenberg ebooks. Uses the
+        strip_headers() function from the gutenberg module.
+
+        :return: string
+        """
+
+        return strip_headers(text.strip())
+
+
+    def remove_table_of_contents(text):
+        """
+        Removes the Table of Contents from an input string of a novel. Written with the intent of
+        being used on Table of Contents removal for Project Gutenberg texts.
+
+        This function works from the observation that books in Project Gutenberg always have
+        chapter titles that are either
+        A) Allcaps words or phrases or
+        B) Capitalized Roman Numerals.
+        This means that the first chapter can be identified as a line which is either
+        A) entirely composed of capital letters or
+        B) entirely composed of the letters {I, V, X, L, C, D, M}
+        [operating under the assumption that no chapter will exceed 3899 chapters, the maximum
+        number that can be obtained with standard Roman Numeral conventions]
+        And that all chapter titles are followed by at least one empty line.
+
+        The function looks for a line which is composed entirely of allcaps or Roman Numerals and
+        followed only by empty lines until finding a line which is not entirely allcaps.
+
+        :return: string
+        TODO(derek): Make this fully functional
+        """
+
+        # Finds all positions of line breaks in the text file for the first few lines, tuned off
+        # the parameter of charactersearch characters out
+        charactersearch = 8000
+        line_breaks = text[0:charactersearch].find("\n")
```

Comment:
This only returns the position of the first line break.

Reply:
re.findall?